doc/_extensions/zephyr/domain/__init__.py

```diff
@@ -213,6 +219,64 @@ def convert_node(self, node):
         node.replace_self(node.children[0])
 
 
+class ConvertBoardNode(SphinxTransform):
+    default_priority = 100
+
+    def apply(self):
+        matcher = NodeMatcher(BoardNode)
+        for node in self.document.traverse(matcher):
+            self.convert_node(node)
+
+    def convert_node(self, node):
+        parent = node.parent
+        siblings_to_move = []
+        if parent is not None:
+            index = parent.index(node)
+            siblings_to_move = parent.children[index + 1 :]
+
+            new_section = nodes.section(ids=[node["id"]])
+            new_section += nodes.title(text=node["full_name"])
+
+            # create a sidebar with all the board details
+            sidebar = nodes.sidebar(classes=["board-overview"])
+            new_section += sidebar
+            sidebar += nodes.title(text="Board Overview")
+
+            if node["image"] is not None:
+                figure = nodes.figure()
+                figure += nodes.image(uri=f"/{node['image']}")
```

Comment:
cool stuff, should the small image be linked to the full size image? we don't really do high res but looking at the page it's tempting to click (not even sure this is the right place to be honest, I'm talking about the board overview card in the board page https://builds.zephyrproject.io/zephyr/pr/79994/docs/boards/seeed/wio_terminal/doc/index.html)

Reply:
Ya it's the right place :) and yes, great suggestion

Reply:
> which I think might actually be a bug in RTD theme, btw, so I'll try to submit a fix upstream as well

https://github.com/readthedocs/sphinx_rtd_theme/pull/1622